packages/frontend/src/system-integration/notifications.ts

```diff
@@ -260,41 +308,87 @@ async function flushNotifications(accountId: number) {
   )
     .filter(e => e.muted)
     .map(e => e.id)
-  if (mutedChats.length > 0) {
-    // some chats are muted
-    log.debug(`ignoring notifications of ${mutedChats.length} muted chats`)
-  }
-  notifications = notifications.filter(notification => {
-    if (mutedChats.includes(notification.chatId)) {
-      // muted chat
-      log.debug('notification ignored: chat muted', notification)
-      return false
-    } else {
-      return true
-    }
-  })
 
-  if (notifications.length > notificationLimit) {
+  const filteredNotifications = (
+    await Promise.all(
+      notifications.map(async notification => {
+        if (mutedChats.includes(notification.chatId)) {
+          // muted chat - only show if it's a mention and mentions are enabled
+          if (SettingsStoreInstance.state?.desktopSettings.isMentionsEnabled) {
+            const isMention = await notificationIsMention(
+              accountId,
+              notification
+            )
+            return isMention ? notification : null
+          }
+          return null
+        } else {
+          return notification
+        }
+      })
```

Comment:
This does not implement the flowchart https://github.com/deltachat/interface/pull/78#issuecomment-2536719734, because it doesn't perform the "is 1:1 chat" check.

Reply:
Yes that's right. So another await Backend call... :-(